core/Tracker/Visit.php

```diff
@@ -130,6 +130,17 @@ public function handle()
 
         $this->visitProperties = new VisitProperties();
 
+        // Break early if request timestamp is older than the cutoff for deleting older log entries
+        $cache = Tracker\Cache::getCacheGeneral();
```

Comment:
👍 works nicely @katebutler 

One thought: let's maybe move the logic to here: https://github.com/matomo-org/matomo/blob/3.x-dev/core/Tracker/Request.php#L503-L505 

Could we also throw an `InvalidRequestParameterException ` and a message that explains that timestamp is too old and data would be deleted again because of log retention ? A `Common::printDebug` be good as well. 

At first was thinking we should fail silently as otherwise anyone could find out whether a user as raw data retention configured and to how many days the raw data retention is set). However, this is not possible because you need to authenticate the request anyway when tracking a request with a timestamp of older than one day. So only users with at least write access would be possible to find out about this and there it is fine. 

Be good to throw an exception 👍